<a href="https://colab.research.google.com/github/maritamayo25/Business-Analytics/blob/main/Reto_2ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Modelo árbol de decisión**
0. Se cargan las librerías de trabajo

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# se trae el modelo
from sklearn.tree import DecisionTreeClassifier

#Para llamar los archivoa del Drive - sin tener que subirlos
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


1. Se cargan los datos desde el drive

In [ ]:

nxl='/content/drive/MyDrive/Analitica de negocios/2. BD2_In_App_Pur_Int.xlsx'
XDB=pd.read_excel(nxl,sheet_name=0)
XDB=XDB. dropna()
XDB.head()

#seleccionamos las variables de interés
XD=XDB[['Age', 'Income', 'Years', 'Hours']]
XD.head()
XD=np.array(XD)#Convierto la tabla en números - Variables de Entrada
yd=XDB[['Buy']]; yd=np.array(yd)    #Variable de salida - Convierto a la tabla

len(XD)                #% de accidentados en la autopista o no
filas=np.where(yd==1)[0]
len(filas)
len(filas)/len(XD)

0.5275

2.Se procede con la construcción del modelo de árbol

In [ ]:
mar=DecisionTreeClassifier(criterion='gini', max_depth=4)
mar.fit(XD,yd)
ydp=mar.predict(XD)#Estos valores de pronóstico sirven para evaluar el modelo

#Evaluamos el modelo
from sklearn.metrics import confusion_matrix
cm=confusion_matrix(yd,ydp)
VN=cm[0,0];  FP=cm[0,1]; FN=cm [1,0]; VP=cm[1,1]

Ex=(VP+VN)/(VP+VN+FP+FN)
Se=VP/(VP+FN); Es=VN/(VN+FP)

print('Exactitud: ',Ex)
print('Sensibilidad: ',Se)
print('Especifidad: ',Es)



Exactitud:  0.8475
Sensibilidad:  0.9478672985781991
Especifidad:  0.7354497354497355


3. Se procede con la visualización del árbol

In [ ]:
from sklearn.tree import export_graphviz #Estructura del Arbol
from pydotplus import graph_from_dot_data #Que datos le voy a colocar al árbol

vs=['Age', 'Income', 'Years', 'Hours'] #Estos nombres aparecen en el árbol
dot_data=export_graphviz(mar,feature_names=vs) #Asegurar nombres del gráfico
graph=graph_from_dot_data(dot_data)                 #Se hace el gráfico del árbol
graph.write_png('arbol.png')
 #Se crea la imaggraphen

True

**Análisis de resultados**

En la base de datos se puede observar un total de 400 jugadores divididos en 211 jugadores que si hacen compras dentro de la  aplicación (53%) y un total de 189 jugadores que no hacen compras dentro de la aplicación. De acuerdo con lo anterior, el modelo de árbol arrojo un indice de exactitud cercano al 85% así como indices de sensibilidad cercanos al 95% igualmente y que muestran el comportamiento del modelo frente a la clasificación de jugadores que hacen compras dentro de la aplicación, Este mismo porcentaje en promedio se puede observar frente al indice de especificidad (74%) el cual mide el comportamiento del modelo en la predicción de jugadores que no harían compras dentro de la aplicación.